modules/gapi/perf/common/gapi_video_perf_tests_inl.hpp

```diff
@@ -241,6 +241,150 @@ PERF_TEST_P_(BackgroundSubtractorPerfTest, TestPerformance)
 
 //------------------------------------------------------------------------------
 
+PERF_TEST_P_(KalmanFilterControlPerfTest, TestPerformance)
+{
+    MatType2 type = -1;
+    int  dDim = -1, mDim = -1;
+    std::size_t testNumMeasurements = 0;
+    cv::GCompileArgs compileArgs;
+    std::tie(type, dDim, mDim, testNumMeasurements, compileArgs) = GetParam();
+
+    const int cDim = 2;
+    cv::gapi::KalmanParams kp;
+    initKalmanParams(kp, type, dDim, mDim, cDim);
+
+    // Generating input
+    std::vector<bool> haveMeasurements;
+    std::vector<cv::Mat> measurements, ctrls;
+    measurements.reserve(testNumMeasurements);
+    ctrls.reserve(testNumMeasurements);
+    haveMeasurements.reserve(testNumMeasurements);
+    cv::RNG& rng = cv::theRNG();
+    {
+        cv::Mat measurement(mDim, 1, type);
+        cv::Mat ctrl(cDim, 1, type);
+        for (std::size_t i = 0; i < testNumMeasurements; i++)
+        {
+            haveMeasurements[i] = rng(2u); // returns 0 or 1 - whether we have measurement
+                                           // at this iteration or not
+            if (haveMeasurements[i] == true)
+            {
+                cv::randu(measurement, cv::Scalar::all(-1), cv::Scalar::all(1));
+                measurements.push_back(measurement);
+            }
+            else
+            {
+                measurements.push_back(cv::Mat::zeros(mDim, 1, type));
+            }
+            cv::randu(ctrl, cv::Scalar::all(-1), cv::Scalar::all(1));
+            ctrls.push_back(ctrl);
+        }
+    }
+
+    // G-API graph declaration
+    cv::GMat m, ctrl;
+    cv::GOpaque<bool> have_m;
+    cv::GMat out = cv::gapi::KalmanFilter(m, have_m, ctrl, kp);
+    cv::GComputation c(cv::GIn(m, have_m, ctrl), cv::GOut(out));
+    auto cc = c.compile(
+        cv::descr_of(cv::gin(cv::Mat(mDim, 1, type), true, cv::Mat(cDim, 1, type))),
+        std::move(compileArgs));
+
+    cv::Mat gapiKState(dDim, 1, type);
+    TEST_CYCLE()
+    {
+        cc.prepareForNewStream();
+        for (std::size_t i = 0; i < testNumMeasurements; i++)
+        {
+            cc(cv::gin(measurements[i], haveMeasurements[i] == true, ctrls[i]),
```

Comment:
Without `== true` and below

Reply:
Here. Here it doesn't work without the workaround. Any ideas why?
Validation is not going well without this

Reply:
@TolyaTalamanov can you take a look on this piece of code?

Reply:
It can be like this:

```
bool v = haveMeasurements[i]
cc(cv::gin(measurements[i], v, ctrls[i]),
```

Reply:
Thanks! applied

Reply:
For those who would wonder: https://riptutorial.com/cplusplus/example/2561/vector-bool---the-exception-to-so-many--so-many-rules